tests/legacy-cli/e2e/setup/002-npm-sandbox.ts

```diff
@@ -17,6 +17,13 @@ export default async function () {
   process.env.NPM_CONFIG_USERCONFIG = npmrc;
   process.env.NPM_CONFIG_PREFIX = npmModulesPrefix;
 
+  // Snapshot builds may contain versions that are not yet released (e.g., RC phase main branch).
+  // In this case peer dependency ranges may not resolve causing npm 7+ to fail during tests.
+  // To support this case, legacy peer dependency mode is enabled for snapshot builds.
+  if (getGlobalVariable('argv')['ng-snapshots']) {
+    process.env['NPM_CONFIG_legacy_peer_deps'] = 'true';
```

Comment:
Can we delete this line from tests/legacy-cli/e2e/utils/packages.ts? I don't see why we'd set it only when invoking `setRegistry` (which some tests invoke to test different npm configs).